### PR TITLE
Fixed issue with write errors, non-detecting objects in multi-index

### PR DESCRIPTION
**Pull request by dimosped**
Fixes issue #645

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## Changelog
 
+### 1.71
+  * Bugfix: #645 Fix write erros for Pandas DataFrame that has mixed object/string types in multi-index column
```
Comment (yschimke) [inline CHANGES.md]:
```suggestion
  * Bugfix: #645 Fix write errors for Pandas DataFrame that has mixed object/string types in multi-index column
```

File: arctic/store/_pandas_ndarray_store.py [hunk 2]
```diff
@@ -183,7 +183,7 @@ def can_write_type(data):
 
     def can_write(self, version, symbol, data):
         if self.can_write_type(data):
-            if np.any(data.dtypes.values == 'object'):
+            if np.any(data.dtypes.values == 'object') or data.index.dtype.hasobject:
```
Comment (yschimke) [inline arctic/store/_pandas_ndarray_store.py]: Should this and below be hasobject also?

Comment (dimosped) [reply arctic/store/_pandas_ndarray_store.py]: "hasobject" can be applied only on a per-dtype basis:

```
In[13]: mydf.dtypes.hasobject
Traceback (most recent call last):
  File "/dev/shm/dpediaditakis/pyenvs/fedex.2.7.15/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 2878, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-13-377951dabaaf>", line 1, in <module>
    mydf.dtypes.hasobject
  File "/dev/shm/dpediaditakis/pyenvs/fedex.2.7.15/lib/python2.7/site-packages/pandas/core/generic.py", line 4376, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'hasobject'
```

An alternative would be:
```
In[14]: np.any([dt.hasobject for dt in mydf.dtypes])
Out[14]: False
```

but I'd rather cause as less regression risks as possible at this stage

File: CHANGES.md (hunk 1 above)
Comment (dimosped) [reply CHANGES.md]: corrected, thanks

File: arctic/store/_pandas_ndarray_store.py (hunk 2 above)
Comment (dimosped) [reply arctic/store/_pandas_ndarray_store.py]: @yschimke  I liked your comment/suggestion, so now the code has homogeneous check for Series object dtype, and added more tests to cover this

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 ## Changelog
 
+### 1.71
```
[the remark this answers is not in the document]
Comment (dimosped) [reply CHANGES.md]: The data has now been added, thanks

**Pull request merged**